server/storage/rethinkdb.go

```diff
@@ -294,7 +324,100 @@ func (rdb RethinkDB) CheckHealth() error {
 	return nil
 }
 
+func (rdb RethinkDB) writeChange(gun string, version int, sha256, category string) error {
+	now := time.Now()
+	ch := Change{
+		CreatedAt: now,
+		GUN:       gun,
+		Version:   version,
+		SHA256:    sha256,
+		Category:  category,
+	}
+	_, err := gorethink.DB(rdb.dbName).Table(ch.TableName()).Insert(
+		ch,
+		gorethink.InsertOpts{
+			Conflict: "error", // default but explicit for clarity of intent
+		},
+	).RunWrite(rdb.sess)
+	return err
+}
+
 // GetChanges is not implemented for RethinkDB
 func (rdb RethinkDB) GetChanges(changeID string, pageSize int, filterName string) ([]Change, error) {
-	return nil, errors.New("Not Implemented")
+	var (
+		lower, upper, bound []interface{}
+		idx                 = "rdb_created_at_id"
+		max                 = []interface{}{gorethink.Now().Sub(blackoutTime), gorethink.MaxVal}
+		min                 = []interface{}{gorethink.MinVal, gorethink.MinVal}
+		order               gorethink.OrderByOpts
+		reversed            bool
+	)
+	if filterName != "" {
+		idx = "rdb_gun_created_at_id"
+		max = append([]interface{}{filterName}, max...)
+		min = append([]interface{}{filterName}, min...)
+	}
+
+	switch changeID {
```

Comment:
would it be helpful to refactor the MySQL `GetChanges` to factor out a helper that handles the `changeID` and `pageSize` logic?  It might be helpful to consolidate that as well, so that the functionality between the two backends is as identical as possible.

Here's the code I had in mind: https://github.com/docker/notary/blob/master/server/storage/sqldb.go#L279-L294

Reply:
Going to leave this for a future PR due to time constraints.